### PR TITLE
[A11y] Better accessibility: checkboxes, buttons & links, visible/invisible content toggles

### LINKED ISSUE #69: Fix DOM nesting warning
```
Warning: validateDOMNesting(...): <a> cannot appear as a descendant of <a>.
    in a (at App.js:32)
    in li (created by Context.Consumer)
    in StyledComponent (created by style__AppSourcesListItem)
    in style__AppSourcesListItem (at App.js:31)
    in ul (created by Context.Consumer)
    in StyledComponent (created by style__AppSourcesList)
    in style__AppSourcesList (at App.js:47)
    in a (created by Context.Consumer)
    in StyledComponent (created by style__AppRowContainer)
    ...
```

A console warning is being thrown because `AppRowContainer` is a link that contains nested links - `AppSourcesListItem`. Maybe everything to the left of `AppSourcesList` should be the "main" link instead of the whole row to prevent conflicts. Low priority - I haven't seen any bugs caused by this beyond the warning.

### LINKED ISSUE #95: Accessibility: show/hide details should be a button

### LINKED ISSUE #96: Accessibility: share buttons should be <a> elements only, not nested buttons

### LINKED ISSUE #97: Accessibility: checkboxes should be checkbox inputs

### LINKED ISSUE #98: Accessibility: add focus styles back across the board

### PR DESCRIPTION
**Pull request by joachimesque**
Closes #95.
Closes #96.
Closes #97.
Closes #98.
Closes #69.

### CODE REVIEW

File: components/ChecklistItem/Apps.js [hunk 1]
```diff
@@ -1,43 +1,74 @@
...
+        </React.Fragment>
       )}
-
-      {overflowAppList &&
-        overflowExpanded &&
-        overflowAppList.map(app => <AppRow key={app.name} app={app} />)}
     </AppsContainer>
   );
-};
+}
```
Comment (joachimesque) [inline components/ChecklistItem/Apps.js]:
```suggestion
};
```

**Comment by brianlovin** [comment]
@joachimesque this is unbelievable, can't believe how thorough this is! Really wonderful attention to detail, and this is just overall so much nicer to use the site now. 💜💜💜💜💜

**Comment by brianlovin** [comment]
Is this ready to ship @joachimesque ?

**Comment by joachimesque** [comment]
Thanks ! Yes it's ready to ship, today I learned the existence of `useRef`, that should explain the late fix to reverse my initial change from a functional component to a Class :)

**Comment by brianlovin** [comment]
Amazing work - let's get this live!

**Pull request merged**